swap/htlc.go

```diff
@@ -190,6 +223,50 @@ func swapHTLCScript(cltvExpiry int32, senderHtlcKey,
 	return builder.Script()
 }
 
+// swapHTLCScriptV2 returns the on-chain HTLC V2 witness script.
+//
+// <receiverHtlcKey> OP_CHECKSIG OP_NOTIF
+//   OP_DUP OP_HASH160 <HASH160(senderHtlcKey)> OP_EQUALVERIFY OP_CHECKSIGVERIFY
+//   <cltv timeout> OP_CHECKLOCKTIMEVERIFY
+// OP_ELSE
+//   OP_SIZE <20> OP_EQUALVERIFY OP_HASH160 <ripemd(swapHash)> OP_EQUALVERIFY 1
+//   OP_CHECKSEQUENCEVERIFY
+// OP_ENDIF
+func swapHTLCScriptV2(cltvExpiry int32, senderHtlcKey,
+	receiverHtlcKey [33]byte, swapHash lntypes.Hash) ([]byte, error) {
+
+	builder := txscript.NewScriptBuilder()
+	builder.AddData(receiverHtlcKey[:])
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	builder.AddOp(txscript.OP_NOTIF)
+
+	builder.AddOp(txscript.OP_DUP)
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(senderHtlcKey[:]))
```

Comment:
Is this the correct hash function?

> the value passed in should be the sha256 of the payment hash

In this case we are revealing the true preimage and not the intermediate state

Reply:
fixed